designer/apps/pages/utils.py

```diff
@@ -0,0 +1,21 @@
+""" Page utils """
+
+
+def is_valid_child_page(cls, parent_child_pages):
+    """
+    When adding a child page (e.g., ProgramPage, EnterprisePage), we want to ensure
+    only child pages of the same type can exist under their parent page.
+
+    For example:
+    - if the parent page has no children, the page would be a valid sibling.
+    - if the parent page already has a child page `ProgramPage`, then only
+      a new `ProgramPage` is valid under the parent page.
+
+    .. no_pii:
+    """
+    is_valid_child = True
```

Comment:
What you have is fine, but dropping this here in case you wanted to save a few lines:
```
    for child_page in parent_child_pages:
        if cls is not child_page.specific_class:
            return False
    return True
```

Reply:
Good suggestion, updated.